pkg/controllers/report/admission/controller.go

```diff
@@ -160,8 +160,17 @@ func (c *controller) reconcile(ctx context.Context, logger logr.Logger, key, nam
 	// cleanup old reports
 	// if they are not the same version as the current resource version
 	// and were created more than five minutes ago
-	if !exists || !reportutils.CompareHash(meta, resource.Hash) {
-		if meta.GetCreationTimestamp().Add(time.Minute * 5).Before(time.Now()) {
+	if !found {
+		// if we didn't find the resource, either no policy exist for this kind
+		// or the resource was never created, we delete the report if it has no owner
+		// and was created more than five minutes ago
+		if len(meta.GetOwnerReferences()) == 0 && meta.GetCreationTimestamp().Add(time.Minute*5).Before(time.Now()) {
```

Comment:
So does this essentially mean it may take up to 5 minutes for a policy report to reflect the current state of the policies in the cluster?

Reply:
Nope, it should be immediate, it means that we have to clean reports at some point because:
- we can create reports for resources that will never get created
- we can create reports for resources that will never get updated
- as resource evolves over time, reports can become obsolete, we don't need to clean them up instantly hence the 5 minutes delay

Shall I reduce the delay here ?

Reply:
Right, I meant "5 minutes to clean a policy report". I understand entries will be added to reports very quickly, but this 5 minute timer means it may take up to 5 minutes to clean them up?

Reply:
In some cases yes but not always. For example if the resource is deleted reports will be garbage collected as usual.

Reply:
I can reduce it, 5 minutes for a resource to be created/updated is large.

Reply:
I'm not able to offer more helpful commentary here so I'll resolve this thread.

Reply:
I reduced the delay to 2 minutes.

Reply:
>we can create reports for resources that will never get created

Is this already handled or this PR deletes the reports for non-existing resources?

Reply:
Sure, this controller is mostly doing cleanup actually as those reports are immutable there's not much to reconcile.